stdlib/3/_curses.pyi

```diff
@@ -292,227 +292,171 @@ def use_env(flag: bool) -> None: ...
 
 class error(Exception): ...
 
-
 class _CursesWindow:
     encoding = ...  # type: str
-
     @overload
-    def addch(self, ch: _chtype, attr: Optional[int]) -> None: ...
-
+    def addch(self, ch: _chtype, attr: int = ...) -> None: ...
     @overload
-    def addch(self, y: int, x: int, ch: _chtype, attr: Optional[int]) -> None: ...
-
+    def addch(self, y: int, x: int, ch: _chtype, attr: int = ...) -> None: ...
     @overload
-    def addnstr(self, str: str, n: int, attr: Optional[int]) -> None: ...
-
+    def addnstr(self, str: str, n: int, attr: int = ...) -> None: ...
     @overload
-    def addnstr(self, y: int, x: int, str: str, n: int, attr: Optional[int]) -> None: ...
-
+    def addnstr(self, y: int, x: int, str: str, n: int, attr: int = ...) -> None: ...
     @overload
-    def addstr(self, str: str, attr: Optional[int]) -> None: ...
-
+    def addstr(self, str: str, attr: int = ...) -> None: ...
     @overload
-    def addstr(self, y: int, x: int, str: str, attr: Optional[int]) -> None: ...
-
+    def addstr(self, y: int, x: int, str: str, attr: int = ...) -> None: ...
     def attroff(self, attr: int) -> None: ...
-
     def attron(self, attr: int) -> None: ...
-
     def attrset(self, attr: int) -> None: ...
-
-    def bkgd(self, ch: _chtype, attr: Optional[int]) -> None: ...
-
-    def bkgset(self, ch: _chtype, attr: Optional[int]) -> None: ...
-
-    def border(self, ls: Optional[int], rs: Optional[int],
-               ts: Optional[int], bs: Optional[int],
-               tl: Optional[int], tr: Optional[int],
-               bl: Optional[int], br: Optional[int]) -> None: ...
-
-    def box(self, vertch: Optional[Tuple[int, int]], horch: Optional[Tuple[int, int]]) -> None: ...
-
+    def bkgd(self, ch: _chtype, attr: int = ...) -> None: ...
+    def bkgset(self, ch: _chtype, attr: int = ...) -> None: ...
+    @overload
```

Comment:
I don't think you need any of these overloads, just the last one is enough. The default of `...` indicates that an argument can be omitted; you only need overloads in special cases like when a function either takes n or n + 2 arguments (like chgat below).

Reply:
The reason I did it the way that I did was from the documentation:
> window.border([ls[, rs[, ts[, bs[, tl[, tr[, bl[, br]]]]]]]])
And ...
>Note:
> A 0 value for any parameter will cause the default character to be used for that parameter. Keyword parameters can not be used.

so all arguments preceding an argument that is needed must be present. So actually I guess I need to remove the '= ...'s to be accurate. 

Though now that I'm thinking about it the fact that they are all the same type then the type checker will not know the difference.

I'll reduce it down to 1 with all optional.